modules/data_integrity_flag/jsx/setFlagForm.js

```diff
@@ -0,0 +1,113 @@
+/* exported SetFlagForm */
+
+class SetFlagForm extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      formData: {}
+    };
+
+    // Bind component instance to custom methods
+    this.setFormData = this.setFormData.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  /**
+   * Set the form data based on state values of child elements/componenets
+   *
+   * @param {string} formElement - name of the selected element
+   * @param {string} value - selected value for corresponding form element
+   */
+  setFormData(formElement, value) {
+    const formData = this.state.formData;
+    formData[formElement] = value;
```

Comment:
Yeah it isn't ideal, won't work for nested objects. I agree that we should use one approach that works in all cases. JSON stringify/parsing sounds good to me. If we run into performance issues later we can check out other solutions like:

https://lodash.com/docs/4.17.4#cloneDeep
https://github.com/substantial/updeep
https://github.com/kolodny/immutability-helper (not really a fan of adding a new syntax like this...)
https://facebook.github.io/immutable-js/